src/types/component-props/parts/contact-option.ts

```diff
@@ -2,28 +2,62 @@ import { PartComponentProps } from '../_component-common';
 import { PartType } from '../parts';
 import { RenderOnAuthStateMixin } from '../_mixins';
 
-export enum ContactOption {
+export enum ChannelType {
     CHAT = 'chat',
     WRITE = 'write',
     CALL = 'call',
 }
 
-export interface ChannelData {
+export interface OpeningHour {
+    status: string;
+    from: string;
+    to: string;
+    dayName?: string;
+    date?: string;
+}
+export interface DefaultContactData {
     ingress?: string;
-    phoneNumber?: string;
     title?: string;
     url?: string;
 }
 
+export interface SharedContactInformationData extends DefaultContactData {
+    sharedContactInformation: {
+        _path: string;
+        data: {
+            contactType: {
+                telephone?: TelephoneData;
+            };
+        };
+    };
+}
+export interface TelephoneData {
+    phoneNumber?: string;
```

Comment:
Kan telefondata mangle telefonnummer?

Reply:
Ja, den kan dessverre det. Rett og slett fordi XP ikke støtter noen form for required i TextLine når den ligger i selve content-type-dataene såvidt jeg kan se av dokumentasjon. Kun regex.